rsc/src/main/java/com/cloudera/livy/rsc/rpc/Rpc.java

```diff
@@ -275,10 +273,14 @@ public void operationComplete(ChannelFuture cf) {
       };
 
       dispatcher.registerRpc(id, promise, msg.getClass().getName());
-      synchronized (channelLock) {
```

Comment:
I think `Channel` is thread safe, it can be called in any thread. The problem here is that we divide the message into two writes, one for header and another for payload, which will mess the ordering of message if called in different threads.

I think there're two solutions to handle it:

1. the one you did, make all the `write` operations execute in one thread, which is channel's `eventloop`. 
2. Another is combining message header and payload together and invoke `write` once. so that we will not get broken message.

The current solution LGTM, thanks for the fix.